ckanext/s3filestore/uploader.py

```diff
@@ -94,7 +94,7 @@ def get_s3_session(config):
 
 def _is_presigned_url(url):
     ''' Determines whether a URL represents a presigned S3 URL.'''
-    parts = url.split('?')
```

Comment:
why not change at 125, just in case we run into another problem due to the binary object.

i.e.
```
 def _cache_get(self, key):
        ''' Get a value from the cache, if enabled, otherwise return None.
        '''
        if not self.signed_url_cache_enabled:
            return None
        redis_conn = connect_to_redis()
        cache_key = _get_cache_key(key)
        return six.text_type(redis_conn.get(cache_key))
```

Reply:
- Code paths where we don't care about the retrieved type can have one less conversion.
- Theoretically we might at some point want to retrieve a bytes value. Redis supports binary types for a reason.

Reply:
should we not have the cache_url be a string also?

https://github.com/qld-gov-au/ckanext-s3filestore/blob/51708821efb33b29f6008e286b9bee7030bdc3e2/ckanext/s3filestore/uploader.py#L275,L279